lightning/bolt12.go

```diff
@@ -0,0 +1,38 @@
+package lightning
+
+/*
+#include "./lib/bolt12/target/bolt12.h"
+#include <stdlib.h>
+*/
+import "C"
+import (
+	"errors"
+	"strings"
+	"unsafe"
+)
+
+type Invoice = C.Invoice
+
+func IsOffer(offer string) bool {
+	return strings.Contains(offer, "lno")
```

Comment:
I think it would be better to actually try to decode instead of using this naive heuristic